### PR TITLE
Refactoring/issue#92

### LINKED ISSUE #92: Add `reduction` parameter to all loss functions and metrics.
**Is your feature request related to a problem? Please describe.**
Some loss functions have this parametr, others don't.
In SSIM and MS-SSIM there is also `size_average` flag which is redundant and copies old PyTorch loss style. 

**Describe the solution you'd like**
Clear interfaces for SSIM, MS-SSIM
Add reduction to VIF and any other metric/loss that don't have it

### PR DESCRIPTION
**Pull request by zakajd**
Closes #92 

## Proposed Changes

  - Removed `size_average` parameter and it's references
  - Adjust tests
  - Rename `reduction_type` parameter in TV into `norm_type`, to avoid synonymous parameters with different meaning

### CODE REVIEW

**Comment by zakajd** [comment]
Ready for review. 
codecov/patch tests is failing and I don't know what to do with that

File: piq/tv.py [hunk 1]
```diff
@@ -108,14 +107,13 @@ def forward(self, prediction: torch.Tensor) -> torch.Tensor:
...
-            score = torch.sum(score, dim=0)
-        elif self.reduction != 'none':
-            raise ValueError(f'Expected "none"|"mean"|"sum" reduction, got {self.reduction}')
-        return score
+        if self.reduction == 'none':
+            return score
+
+        return {'mean': score.mean,
+                'sum': score.sum
+                }[self.reduction](dim=0)
```
Comment (denproc) [inline piq/tv.py:119]: I suggest to move it to `forward`. The function returns reduced values already, therefore the processing of the reduction is not needed here.

Comment (zakajd) [reply piq/tv.py]: Ready 👍

**Comment by zakajd** [comment]
codecov is broken again :(

**Comment by denproc** [comment]
Looks like known [issue of codecov](https://github.com/codecov/codecov-action/issues/37). Some say it was fixed a couple days ago, but we still experience the problem.
